pwiz/data/vendor_readers/Waters/ChromatogramList_Waters.cpp

```diff
@@ -329,9 +353,49 @@ PWIZ_API_DECL void ChromatogramList_Waters::createIndex() const
         }
     }
 
+    long analogChannels = rawdata_ -> AnalogIntensitiesByChannel().size();
+
+    for(int ch=0; ch < analogChannels; ch++)
+    {
+        index_.push_back(IndexEntry());
+        IndexEntry& ie = index_.back();
+        ie.index = index_.size() - 1;
+        ie.function = -1;
+        ie.analogChannel = ch;
+        ie.offset = 0;
+        ie.Q1 = 0;
+
+        std::string temp = rawdata_->AnalogChannelNames()[ch];
+
+        trim((temp));
+        ie.id = temp;
+        ie.chromatogramType = MS_Analog_chromatogram;

```

Comment:
As mentioned above, chromatogramType should be a specific type (like EMR chromatogram or flow rate chromatogram). If the type isn't known, then use `MS_chromatogram_type`.

Also, the id must be unique. Do we know for sure whether AnalogChannelNames are unique?